framework/tlvf/src/src/tlvftypes.cpp

```diff
@@ -0,0 +1,81 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include <climits>
+#include <linux/if_ether.h>
+#include <tlvf/tlvftypes.h>
+
+namespace tlvf {
+std::string int_to_hex_string(const unsigned int integer, const uint8_t number_of_digits)
+{
+    // 'number_of_digits' represent how much digits the number should have, so the function will
+    // pad the number with zeroes from left, if necessary.
+    // for example: int_to_hex_string(255, 4) -> "00ff"
+    //              int_to_hex_string(255, 1) -> "ff"
+
+    std::stringstream ss_hex_string;
+
+    // convert to hex
+    ss_hex_string << std::setw(number_of_digits) << std::setfill('0') << std::hex << integer;
+
+    return ss_hex_string.str();
+};
+
+// Converts a mac address in a human-readable format
+std::string mac_to_string(const uint8_t *mac_address)
+{
+    std::string mac_addr_string;
+
+    mac_addr_string = int_to_hex_string((uint32_t)mac_address[0], 2) + ":" +
```

Comment:
Define a constexpr "separator" character instead of repeating ":" all along this file

Reply:
I don't really see the benefit for doing this cause it is used only in two functions, once as char * and once as a char, but OK.